tokio/src/runtime/time/entry.rs

```diff
@@ -450,16 +415,16 @@ impl TimerShared {
 }
 
 unsafe impl linked_list::Link for TimerShared {
-    type Handle = TimerHandle;
+    type Handle = NonNull<TimerShared>;
 
     type Target = TimerShared;
 
     fn as_raw(handle: &Self::Handle) -> NonNull<Self::Target> {
-        handle.inner
+        *handle
     }
 
     unsafe fn from_raw(ptr: NonNull<Self::Target>) -> Self::Handle {
-        TimerHandle { inner: ptr }
+        ptr
     }
```

Comment:
Why are you moving away from `TimerHandle` here?

Reply:
Because of https://github.com/tokio-rs/tokio/blob/833ee027d0ec44d88765157ec98b6809f0070169/tokio/src/util/linked_list.rs#L354-L358